### PR TITLE
Ft nag implementation

### PR DESCRIPTION
**Pull request by FelixWambiri**
#### What does this PR do?
Nag user by email until they have resolved an issue and introduce a nag icon
#### Description of Task to be completed?
A user can configure nag user intervals
A user can visually see that a check is in nag mode

#### How should this be manually tested?
After clowning the repo,cd into it and run: 

- python manage.py makemigrations 
- python manage.py migrate
- python manage.py test

#### Screenshot

screen shot 2018-03-29 at 08 38 39
screen shot 2018-03-29 at 08 39 05
screen shot 2018-03-29 at 08 41 38

### CODE REVIEW

File: templates/front/docs.html [hunk 1]
```diff
@@ -244,6 +244,17 @@ <h2>When Alerts Are Sent</h2>
             sends you an alert.
         </td>
     </tr>
```
Comment (meshnesh) [inline templates/front/docs.html:246]: Good implementation of the nag feature

File: static/js/checks.js [hunk 2]
```diff
@@ -93,6 +93,34 @@ $(function () {
         $("#update-timeout-grace").val(rounded);
     });
 
+    var nagSlider = document.getElementById("nag-slider");
```
Comment (meshnesh) [inline static/js/checks.js:96]: Good implementation of the JS language in the UI

File: hc/front/tests/test_my_checks.py [hunk 3]
```diff
@@ -58,3 +58,14 @@ def test_it_shows_amber_check(self):
 
         # Mobile
         self.assertContains(r, "label-warning")
+
+    def test_it_shows_nag_check(self):
```
Comment (meshnesh) [inline hc/front/tests/test_my_checks.py]: it's good practice to add docstring on tests so as they explain better what the test achieves

File: hc/api/management/commands/sendalerts.py [hunk 4]
```diff
@@ -21,8 +21,10 @@ def handle_many(self):
         now = timezone.now()
         going_down = query.filter(alert_after__lt=now, status="up")
         going_up = query.filter(alert_after__gt=now, status="down")
+        nag_down = query.filter(nag_after__lt=now, status="down")
+        nag_up = query.filter(nag_after__gt=now, status="nag")
```
Comment (Kalela) [inline hc/api/management/commands/sendalerts.py]: We can add nag functionality for my story on jobs running too often after the demo.

**Review by Kalela — APPROVED** [review]

File: hc/api/management/commands/sendalerts.py (hunk 4 above)
Comment (betsybeth) [reply hc/api/management/commands/sendalerts.py]: Good implementation of nag slider

**Comment by betsybeth** [comment]
Good work on adding docstring and comments

File: static/js/checks.js (hunk 2 above)
Comment (betsybeth) [reply static/js/checks.js:96]: Good implementation of  UIslider

**Review by lakivisi-zz — CHANGES_REQUESTED** [review]
Improve on coverage. Check coveralls to see non-tested lines

File: hc/api/models.py [hunk 5]
```diff
@@ -92,7 +96,12 @@ def get_status(self):
 
         if self.last_ping + self.timeout + self.grace > now:
             return "up"
-
+        elif self.last_ping + self.timeout + self.grace + self.nag < now:
```
Comment (meshnesh) [inline hc/api/models.py:99]: nice implementation

File: hc/front/tests/test_update_timeout.py [hunk 6]
```diff
@@ -79,4 +79,26 @@ def test_update_timeout_and_grace_works(self):
         check = Check.objects.get(code=self.check.code)
         assert check.timeout.total_seconds() == 300 
         assert check.grace.total_seconds() == 90 
-              
+            
+
+def test_update_timeout_and_nag_works(self):
```
Comment (frankip) [inline hc/front/tests/test_update_timeout.py]: good logical data flow

File: templates/front/my_checks_mobile.html [hunk 7]
```diff
@@ -35,6 +35,8 @@ <h2>
                         <span class="label label-danger">DOWN</span>
                     {% elif check.get_status == "often" %}
                         <span class="label label-info">OFTEN</span>
+                    {% elif check.get_status == "nag" %}
+                        <span class="label label-nag">Nag</span>
```
Comment (frankip) [inline templates/front/my_checks_mobile.html:39]: nice implementation

File: templates/front/my_checks_desktop.html [hunk 8]
```diff
@@ -29,7 +29,7 @@
             {% elif check.get_status == "up" %}
                 <span class="status icon-up"></span>
             {% elif check.get_status == "down" %}
-                <span class="status icon-down"></span>
+               <span class="status icon-down"></span>
```
Comment (frankip) [inline templates/front/my_checks_desktop.html:32]: what change did you make here they looks the same

File: templates/emails/alert-subject.html [hunk 9]
```diff
@@ -1,2 +1,6 @@
+{% if check.status == 'nag' %}
```
Comment (frankip) [inline templates/emails/alert-subject.html:1]: nice work around

**Review by frankip — COMMENTED** [review]
you may want to group some of the small edits you do to one commit message, you have a lot of commits

File: hc/front/tests/test_my_checks.py [hunk 10]
```diff
@@ -66,11 +66,18 @@ def test_it_shows_purple_check(self):
         self.client.get("/ping/%s/" % self.check.code)
         self.check.refresh_from_db()
 
+    def test_it_shows_nag_check(self):
+        self.client.login(username="alice@example.org", password="password")
+
+        self.check.last_ping = timezone.now() - td(days=3)
+        self.check.status = "nag"
+        self.check.save()
+
```
Comment (Kalela) [inline hc/front/tests/test_my_checks.py:75]: Great job adding tests for better coverage.

Comment (betsybeth) [reply hc/front/tests/test_my_checks.py:75]: Good work on writing  tests

**Review by lakivisi-zz — APPROVED** [review]

**Comment by betsybeth** [comment]
Good work

**Pull request merged**